apps/testingaccessibility/src/pages/api/certificate/modules/[slug].ts

```diff
@@ -0,0 +1,214 @@
+
+      const moduleImage = await fetchImage(module.image.url)
+      const imageSVG = canvas
+        .image(moduleImage)
+        .width(320)
+        .height(320)
+        .attr('x', '36%')
+        .attr('y', '6%')
+
+      const userName = req.query.name || sessionToken.name || ''
+      const nameSVG = canvas
+        .text(userName)
+        .attr('font-size', 36)
+        .attr('x', '50%')
+        .attr('y', '66%')
+        .attr('text-anchor', 'middle')
+        .fill('#000')
+
+      const completedEverythingSVG = canvas
+        .text(
+          toUpper(`Has successfully completed Testing Accessibility workshops`),
+        )
+        .attr('font-size', 18)
+        .attr('x', '50%')
+        .attr('y', '74%')
+        .attr('text-anchor', 'middle')
+        .fill('#000')
+
+      const completedModuleSVG = canvas
+        .text(toUpper(`Has successfully completed`))
+        .attr('font-size', 18)
+        .attr('x', '50%')
+        .attr('y', '74%')
+        .attr('text-anchor', 'middle')
+        .fill('#000')
+
+      const completedModuleRestSVG = canvas
+        .text(toUpper(`the ${module.title} workshop`))
+        .attr('font-size', 18)
+        .attr('x', '50%')
+        .attr('y', '77%')
+        .attr('text-anchor', 'middle')
+        .fill('#000')
+
+      // default font doesnt support all characters so we need to load a custom one
+      SVGtoPDF(doc, nameSVG.svg(), 0, 0, {
+        fontCallback: () => path.resolve('./public/fonts/Inter-Bold.otf'),
+      })
+
+      if (slug) {
+        SVGtoPDF(doc, imageSVG.svg(), 0, 0)
+        SVGtoPDF(doc, completedModuleSVG.svg(), 0, 0)
+        SVGtoPDF(doc, completedModuleRestSVG.svg(), 0, 0)
+      } else {
+        SVGtoPDF(doc, completedEverythingSVG.svg(), 0, 0)
+      }
+
+      res.setHeader(
+        'Content-disposition',
+        "attachment; filename*=UTF-8''" +
+          `Testing Accessibility Certificate - ${encodeURI(
+            userName as string,
+          )}.pdf`,
+      )
+      res.setHeader('Content-type', 'application/pdf; charset=UTF-8')
+      doc.pipe(res)
+      doc.end()
+    } catch (error: any) {
+      Sentry.captureException(error)
+      res.status(500).json({error: true, message: error.message})
+    }
+  } else {
+    console.error('non-get request made')
+    res.status(404).end()
+  }
+}
+
+export default withSentry(generateModuleCertificate)
+
+export const config = {
+  api: {
+    externalResolver: true,
+  },
+}
+
+const backgroundPath = path.resolve(
+  './public/assets/certificate-background.svg',
+)
+const background = fs.readFileSync(backgroundPath).toString('utf-8')
+
+const allModulesQuery = groq`*[_type == "module"]{
+    "slug": slug.current
+}`
+
+const moduleQuery = groq`*[_type == "module" && slug.current == $slug][0]{
+title,
+"slug": slug.current,
+image {
+    url
+},
+sections[]->{
+      title,
+      "slug": slug.current,
+      lessons[]->{
+        title,
+        "slug": slug.current
+      }
+}
+}`
```

Comment:
query's like this should probably be moved to the `lib` folder in a `modules.ts` or similar:

https://github.com/skillrecordings/products/blob/27357bbc57be3417bdf376ed71fc673f29e30616/apps/testingaccessibility/src/lib/articles.ts

basically ANY external api calls should be in the lib folder